hal/src/sercom/v2/uart/reg.rs

```diff
@@ -271,8 +290,16 @@ impl<S: Sercom> Registers<S> {
         use BaudMode::*;
         use Oversampling::*;
 
+        #[cfg(feature = "samd20")]
```

Comment:
since this is so littered with `#[cfg]`s, I would consider separate functions. Or maybe config blocks, like
```rust
#[cfg(feature = "samd20")]
{
    // all of the samd20 logic
}

#[cfg(feature = "samd21")]
{
    // all of the samd21 logic
}
```

Reply:
Oh, perfect, I thank putting them in blocks would have made the variables definitions in another scope. That's fixed :)

Reply:
As I said in an earlier comment, it would be better to put all the SAMD20-specific functions in their own `impl` block, as well as all the functions disabled on SAMD20 in their own impl. That means you would only need two `#[cfg]s` and it would make the whole thing much more readable :) There would be two separate `set_baud` and `get_baud` functions in this case.